### PR TITLE
Chroe: add agent docker images release to ghcr registry

### PR DESCRIPTION
**Pull request by JaredTan95**

### CODE REVIEW

File: .github/workflows/release-dev-images.yaml [hunk 1]
```diff
@@ -11,14 +11,18 @@ on:
   push:
     branches:
       - main
+concurrency:
+  group: starship-${{ github.event.pull_request.number || github.ref }}
```
Comment (nascentcore-eng) [inline .github/workflows/release-dev-images.yaml:17]: Do we need this?

Comment (nascentcore-eng) [inline .github/workflows/release-dev-images.yaml:17]: What's the purpose of this config? Please document in the comment

File: .github/workflows/release-dev-images.yaml [hunk 2]
```diff
@@ -11,14 +11,18 @@ on:
...
+  cancel-in-progress: true
 
 env:
   REGISTRY_SERVER_ADDRESS: ghcr.io/${{ github.repository_owner }}
   REGISTRY_USER_NAME: ${{ github.repository_owner }}
   REGISTRY_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+  AWS_REGION: ap-northeast-1
```
Comment (nascentcore-eng) [inline .github/workflows/release-dev-images.yaml]: What's the purpose of this config? Please document in the comment

Comment (nascentcore-eng) [reply .github/workflows/release-dev-images.yaml]: I think you probably don't need this.

**Review by nascentcore-eng — COMMENTED** [review]
Please run the agent image locally and make sure the linux header are installed correctly.

**Comment by nascentcore-eng** [comment]
> Please run the agent image locally and make sure the linux header are installed correctly.

Please include the test results as screenshot in the comment for record.

File: src/agent/cmd/BUILD.bazel [hunk 3]
```diff
@@ -41,10 +42,16 @@ go_image(
...
+    name = "linux_headers",
+    srcs = ["@download_linux_headers_from_s3_url//file"],
+    package_dir = "/",
+)
+
 container_image(
     name = "agent_image",
     base = ":agent_base_image",
-    tars = ["//devops/linux_headers/build:download_linux_headers_from_s3"],
+    tars = [":linux_headers"],
```
Comment (oojimmy) [inline src/agent/cmd/BUILD.bazel]: Maybe mount dir is not right?

**Review by oojimmy — CHANGES_REQUESTED** [review]

**Comment by JaredTan95** [comment]
tested the new agent image packaged by CI:
image

**Pull request merged**